pkg/sql/sem/plpgsqltree/statements.go

```diff
@@ -0,0 +1,541 @@
+}
+
+// stmt_for
+type PLpgSQLStmtForIntLoop struct {
+	PLpgSQLStatementImpl
+	Label   string
+	Var     PLpgSQLVariable
+	Lower   PLpgSQLExpr
+	Upper   PLpgSQLExpr
+	Step    PLpgSQLExpr
+	Reverse int
+	Body    []PLpgSQLStatement
+}
+
+func (s *PLpgSQLStmtForIntLoop) Format(ctx *tree.FmtCtx) {
+}
+
+type PLpgSQLStmtForQueryLoop struct {
+	PLpgSQLStatementImpl
+	Label string
+	Var   PLpgSQLVariable
+	Body  []PLpgSQLStatement
+}
+
+func (s *PLpgSQLStmtForQueryLoop) Format(ctx *tree.FmtCtx) {
+}
+
+type PLpgSQLStmtForQuerySelectLoop struct {
+	PLpgSQLStmtForQueryLoop
+	Query PLpgSQLExpr
+}
+
+func (s *PLpgSQLStmtForQuerySelectLoop) Format(ctx *tree.FmtCtx) {
+}
+
+type PLpgSQLStmtForQueryCursorLoop struct {
+	PLpgSQLStmtForQueryLoop
+	CurVar   int // TODO is this CursorVariable?
+	ArgQuery PLpgSQLExpr
+}
+
+func (s *PLpgSQLStmtForQueryCursorLoop) Format(ctx *tree.FmtCtx) {
+}
+
+type PLpgSQLStmtForDynamicLoop struct {
+	PLpgSQLStmtForQueryLoop
+	Query  PLpgSQLExpr
+	Params []PLpgSQLExpr
+}
+
+func (s *PLpgSQLStmtForDynamicLoop) Format(ctx *tree.FmtCtx) {
+}
+
+// stmt_foreach_a
+type PLpgSQLStmtForEachALoop struct {
+	PLpgSQLStatementImpl
+	Label string
+	Var   *PLpgSQLVariable
+	Slice int // TODO not sure what this is
+	Expr  PLpgSQLExpr
+	Body  []PLpgSQLStatement
+}
+
+func (s *PLpgSQLStmtForEachALoop) Format(ctx *tree.FmtCtx) {
+}
+
+// stmt_exit
+type PLpgSQLStmtExit struct {
+	PLpgSQLStatementImpl
+	IsExit    bool
+	Label     string
+	Condition PLpgSQLExpr
+}
+
+func (s *PLpgSQLStmtExit) Format(ctx *tree.FmtCtx) {
+	ctx.WriteString("EXIT\n")
+}
+
+// stmt_return
+type PLpgSQLStmtReturn struct {
+	PLpgSQLStatementImpl
+	Expr   PLpgSQLExpr
+	RetVar PLpgSQLVariable
+}
+
+func (s *PLpgSQLStmtReturn) Format(ctx *tree.FmtCtx) {
+}
+
+type PLpgSQLStmtReturnNext struct {
+	PLpgSQLStatementImpl
+	Expr   PLpgSQLExpr
+	RetVar PLpgSQLVariable
+}
+
+func (s *PLpgSQLStmtReturnNext) Format(ctx *tree.FmtCtx) {
+}
+
+type PLpgSQLStmtReturnQuery struct {
+	PLpgSQLStatementImpl
+	Query        PLpgSQLExpr
+	DynamicQuery PLpgSQLExpr
+	Params       []PLpgSQLExpr
+}
+
+func (s *PLpgSQLStmtReturnQuery) Format(ctx *tree.FmtCtx) {
+}
+
+// stmt_raise
+type PLpgSQLStmtRaise struct {
+	PLpgSQLStatementImpl
+	LogLevel int
+	CodeName string
+	Message  string
+	Params   []PLpgSQLExpr
+	Options  []PLpgSQLStmtRaiseOption
+}
+
+func (s *PLpgSQLStmtRaise) Format(ctx *tree.FmtCtx) {
+}
+
+type PLpgSQLStmtRaiseOption struct {
+	OptType PLpgSQLRaiseOptionType
+	Expr    PLpgSQLExpr
+}
+
+func (s *PLpgSQLStmtRaiseOption) Format(ctx *tree.FmtCtx) {
+}
+
+// stmt_assert
+type PLpgSQLStmtAssert struct {
+	PLpgSQLStatementImpl
+	Condition PLpgSQLExpr
+	Message   PLpgSQLExpr
+}
+
+func (s *PLpgSQLStmtAssert) Format(ctx *tree.FmtCtx) {
+	ctx.WriteString("ASSERT\n")
+}
+
+// stmt_execsql
+type PLpgSQLStmtExecSql struct {
+	PLpgSQLStatementImpl
+	SqlStmt            string
+	IsModifyingStmt    bool            // is the stmt insert/update/delete/merge?
```

Comment:
We should reword these comments to be not questions but explanations :) But I think we should remove the field not used for now cause it might be confusing why they exist.